app/uk/gov/hmrc/agentsubscriptionfrontend/connectors/AgentAssuranceConnector.scala

```diff
@@ -55,6 +57,19 @@ class AgentAssuranceConnector @Inject()(@Named("agent-assurance-baseUrl") baseUr
     }
   }
 
+  def isR2DWAgent(utr: Utr)(implicit hc: HeaderCarrier): Future[Boolean] = {
```

Comment:
What about having an endpoint to check by UTR and receive 204/404? It would be more efficient then ask for a list each time.

Reply:
this followed the instructions from the use case, although we had a discussion with Martin and we will be changing this to a yes/no endpoint